### PR TITLE
CASMINST-3831: Regenerate the the host_records metadata using data from SLS during CSM 1.2 upgrade 

### PR DESCRIPTION
**Pull request by rsjostrand-hpe**
#### Summary and Scope
- Fixes # [CASMINST-3931](https://jira-pro.its.hpecorp.net:8443/browse/CASMINST-3831)
* When updating BSS metadata re-generate the the host_records metadata using data from SLS for the Global cloud-init metadata. Basically I reimplemented the [pit.MakeBasecampHostRecords](https://github.com/Cray-HPE/cray-site-init/blob/main/pkg/pit/basecamp.go#L159) to use data from SLS. Meaning that BSS will contain the expected that a fresh install would contain after an upgrade.
  * CAN IP addresses are correctly updated
  * CMN and CHN host records are now present
  * Switch host_records are now using the CSM 1.2 names
* Added the CHN to the list of BSS IPAM networks
* Removed the longer needed `can-if` and `can-gw` cloud-init metadata parameters.

##### Issue Type
- Bugfix Pull Request

#### Prerequisites

- [x] I have included documentation in my PR (or it is not required)
- [ ] I tested this on internal system (x) (if yes, please include results or a description of the test)
 
#### Idempotency
 The `csi upgrade metadata --1-0-to-1-2` command should be idempotent if it is ran multiple times during a CSM upgrade.  The cloud-init host_record data is recomputed from SLS data each time it is built.  
 
#### Testing
Only local testing was used to verify these changes are correct. 

The `csi upgrade metadata --1-0-to-1-2` command is destructive and will modify the contents of BSS, and for a full test I need a system that has SLS update with CSM 1.2 data, but BSS has not modified yet. So testing for these changes are being done by the added unit tests. 

The mock data upgrade metadata unit tests are based on Fanta. 
* BSS Bootparameters for a CSM 1.0 system. Collected from Fanta running CSM 1.0.
* SLS data for a CSM 1.2 system. Collected from generating CSM 1.2 configs for Fanta using CSI
* Expected BSS Global Bootparameters for a CSM 1.2 system. Collected from CSM 1.2 configs for Fanta  using CSI. 

Two unit tests have been added:
1. The function `getBSSGlobalHostRecords` has unit tests to verify that it generates the expected host record list from SLS data.
2. The function `updateBSS_oneToOneTwo` has unit tests to verify that the Global BSS Bootparameters structure was correctly modified for CSM 1.2.

#### Risks and Mitigations
Requires additional end to end testing on Baremetal during a CSM 1.2 upgrade. But all of the changes made to a system are contained within the Global boot parameters in BSS, which currently contains incorrect host record data after a CSM 1.2 upgrade.

### CODE REVIEW

**Comment by rsjostrand-hpe** [comment]
Here is what the updated Global boot parameters look like out of the unit test:
```
{
  "hosts": [
    "Global"
  ],
  "params": "metal.no-wipe=1",
  "cloud-init": {
    "meta-data": {
      "ca-certs": {
        "remove-defaults": false,
        "trusted": [
          "-----BEGIN CERTIFICATE-----\nMIIEuDCCAyCgAwIBAgIUC6dq8AfZ9WeMP96x8+hYcrYbWRwwDQYJKoZIhvcNAQEL\nBQAwYTEPMA0GA1UECgwGU2hhc3RhMREwDwYDVQQLDAhQbGF0Zm9ybTE7MDkGA1UE\nAwwyUGxhdGZvcm0gQ0EgKDAzMjQyMWMyLWMwNjctNDU3OC1iNWRjLTljMjU1MTVl\nZTMyNykwHhcNMjIwMTEyMDUzOTUyWhcNMzIwMTEwMDUzOTUyWjBmMQ8wDQYDVQQK\nDAZTaGFzdGExETAPBgNVBAsMCFBsYXRmb3JtMUAwPgYDVQQDDDdQbGF0Zm9ybSBD\nQSAtIEwxICgwMzI0MjFjMi1jMDY3LTQ1NzgtYjVkYy05YzI1NTE1ZWUzMjcpMIIB\nojANBgkqhkiG9w0BAQEFAAOCAY8AMIIBigKCAYEAwyMKwD6ub1YYGqbxibWRnuid\n65Kvtv5NtHCj79DJj4AtptBSnxB6hZVlj5EaTdkpRotD2b46e74b4zPsoJKJKegT\nJheUOLSwqj0u9xwPeqqO/enk60eY9gJHWIkq9W3Bif50elEKMeksxv94u8kcYJH/\nzeEAbOLeY/cV60bwpcKoQm1ztCmgeGWJxxYM85Mt0d+SmvsMvFm/dCsmdsXiZtca\nd6EqG8FhDmCKv/8o2xFRPjxj9OjssRfOFDE41IEe/GcklvJ4lxhcqlRsMhDXr8/e\nhpEoynKxRZIXo6bSatkm6K7IadDYNyIrDfbnt1CJxCtVLECLzo+Xht/hHQaHRk5F\nlesIbEUUBiHFWHHI51oL9GgDuRa5DpKqNn7xpTYPt4LZ0sZhX4E1C2QePoP8kfgQ\nc1s8ZqCY2iVH/rWv0EHsab9BoFH9dRMnep7J9EePSFoSqWACJKZ3f0gR4QIWrOei\nlMfxYZgG5Zr7Nt1nGbdl48n7CSrN6wxxSXVcVCW3AgMBAAGjYzBhMA8GA1UdEwEB\n/wQFMAMBAf8wDgYDVR0PAQH/BAQDAgEGMB0GA1UdDgQWBBSIiI1jVwVdQOKunZxx\ngW7ZJBfWGTAfBgNVHSMEGDAWgBROirlRRwKPQagW/95nmDFIrkFfojANBgkqhkiG\n9w0BAQsFAAOCAYEAe1YnegAPsP6pb1z9g7rjPfrlGhLoIDL5XDtpnjPh0/kdJnV+\nVh42emTgiI8X0oO6nzPbiJyLcnsSsfN1Lv/SMWrTOtDwNPBWz5dOXNQQoc43tWPP\nbSbNe/ncHZo8MdQNrHrN9+Q5wer3o9BXwSQ7Cl847DvN0z0UhxNK+HqO6MRJnlge\nUTkFm7lelg/auQsbH+GIkRtnWrrl2MKNxqFrUu+LTGsq3yGcjyCqlY7G9kGggVky\nbPEnOUedQIIX0PhEg2KAdAKNecKziCTDoyNzSX6Kyqb3CWDFzxAB7Or1FrDPqYF5\nu84L86N57qStGqQBRuPkA9b5YEM6ZyjrZXVv1Nu/s5r0mhiTOaqLuSpVy/d4H4Ly\n7ISQtXTQw3UmfRNPNl3UzEQKHoWbDArvIq9BBrhcRZyLV4aB/jX0X6gRsNmmWLnD\nmUYrj1nFKWEWw26zNImiz1nR2yJj40XSwId8L6njkJJEuQ+IPAWi3a1kduKx+Ty6\n+y5PjjZ0/3CEH9b6\n-----END CERTIFICATE-----\n",
          "-----BEGIN CERTIFICATE-----\nMIIEkjCCAvqgAwIBAgIULXDChgvCqdBbevxsQneb7OztU6MwDQYJKoZIhvcNAQEL\nBQAwYTEPMA0GA1UECgwGU2hhc3RhMREwDwYDVQQLDAhQbGF0Zm9ybTE7MDkGA1UE\nAwwyUGxhdGZvcm0gQ0EgKDAzMjQyMWMyLWMwNjctNDU3OC1iNWRjLTljMjU1MTVl\nZTMyNykwHhcNMjIwMTEyMDUzOTUyWhcNMzIwMTExMDUzOTUyWjBhMQ8wDQYDVQQK\nDAZTaGFzdGExETAPBgNVBAsMCFBsYXRmb3JtMTswOQYDVQQDDDJQbGF0Zm9ybSBD\nQSAoMDMyNDIxYzItYzA2Ny00NTc4LWI1ZGMtOWMyNTUxNWVlMzI3KTCCAaIwDQYJ\nKoZIhvcNAQEBBQADggGPADCCAYoCggGBANBNe3J8SsoqWekfz05tkE4NE272E/vv\nQamvEyryRbtBZPTSfD9vUjIXZ32kMwvBB+DfpH0/qTs5FX5Ma6bOi0643WeZ3ue/\nJJp6/hIQLdoNFPVE4pMJIA7A1Yv1swP9oZt3vCB9goElQiYHWj+nDUMPwJUBhXy3\nBzZzTKWgfLVp0Aw4GUbVaECv/+K4AP8t5fX1FA24ICQnRMpaE0T/nnVqfAtaEUYN\nw7eX8cKrNyJw35XVkTJ9ToKOpEFQ9lu3dlIO+M0N3DXqIkFHcZjvEzdiIdYymF5X\nAkMBnA3derjrJmIocnP+KMMv3RGI+m6gViM7qzuXMuHRd63tjH2xSx4ItJP5mH8l\n5PNZrFUf0GE030uRyOtHZNSuGUx1cMtAawUPSj+XQtcvpg5AoajHWXYFt0DItVjE\n6NgOQlvEPC3oIenmaJF8A4+09I+f1KVWmEZsSuN8znS5TImt88sMkQ+lt1+JsOgz\nGDB06RdJxg28qE6a3JaYYBObFkFg2ZhnkQIDAQABo0IwQDAPBgNVHRMBAf8EBTAD\nAQH/MA4GA1UdDwEB/wQEAwIBBjAdBgNVHQ4EFgQUToq5UUcCj0GoFv/eZ5gxSK5B\nX6IwDQYJKoZIhvcNAQELBQADggGBAGHX/Ddky4n0cWHc8zYgu8D7FpAVn84fgu9p\noSKt1WRKb72Czwel5NOyP8rxoR6jkUY5SMBtONz9yQu/uqujJ4DRhQEaZqc/gIKU\nBbPu/VUntQVkZY/vNT/fbV1E6JvC8dn6gFwIsEJHTrZlroo6njIYP3b3lmNyBNl+\noAXFpkuSOkB5OczI+IcK2SvnxuWWd6DDN5KIMBt1jDQ9xqjXBtLLw0BQycIqqriD\nYaXvYwyiDwDUOZUMQUbx1fB819UPnF3hVUUo9NiaIdukDMAy6sW0gZBdm8Z6eW39\n5M8oDaVa+PoNqe6MVWN8OjWu1QDG7i1VqlATODA+p1Pqf3QW7jk5XE14IAm57C9N\nFN0CsEtQOpb/HHTp8zxzVzme9M5CiXo4pA41r3q1mKAqRmbeyeqfzmR5NOJNGUNI\nl868tqp2Z7Xdsk77puPnUbFlk3102EuAB1DsvBne3A7MB1fNLRyGL3opFAecpv1k\nmlaw3YIPzBQvu3TmYHHW6wRt1+gfGw==\n-----END CERTIFICATE-----\n"
        ]
      },
      "ceph-cephfs-image": "dtr.dev.cray.com/cray/cray-cephfs-provisioner:0.1.0-nautilus-1.3",
      "ceph-rbd-image": "dtr.dev.cray.com/cray/cray-rbd-provisioner:0.1.0-nautilus-1.3",
      "cfs_public_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDLpgxwAoRztlUoBUKwrHsXbMpWcg4bPLxZ2JfsJUFijJN7ai7vzEB08oHYyeoxJ2u2+9LUN7DgARu3C7Fj/Zc5NLvqOIBIvLE8zec5j0wT7qiaJKZgaSMaAlbOmVUcPGP6joQXIZ7q/YMJp4jBy0oZMZagxZu0vFeKKi8Y4RYgI9RqncY+rcrAqdFJBiHUdbuMGGFHLx7nhio08VccdN3BYseYsqRlRurpFJ5mUW+CLbXAjD/IfhdYAkjU5cvTNIIsH09ciQ3JBuxqguNNTCtF9wvXvYJiFSUwE1BU5Tz8iTwAiUpx72X0L+52f+YXs3BGKzXW+HrXfzEX8dmkl1Kxyrm9hUorcohuKh4h6MTXFMuerD9NGHCpkLCy02fGEqEBS23uH1CZm+dsAGEtKHgrf5MDfEnmYIypHNNgk2VQJfwAbUNqS8SqBU1S3YJDHNX0N4D3K48LcZhRU02KPzkNNuTkclix8JIdaBs/QFPCSNL8OL1A5mQ9A+P2e6lI3inuFrqIjvqXvRpM6h/QVuipYFuZN7N7mEvR68b/uWIUotTSZNPixJjcLIaG8vDU+B5kJ+6rWXUiawCIHOGHzdkRTZgEqPVdq4qUafWhWdX9SrfQfdOehdFKo9hyXTAORW9+eKrsJmMeAsKzXQZ6zeYQrmPuV5CuEq1E8vKmfZ3QkQ==\n",
      "dns-server": "10.92.100.225",
      "docker-image-registry": "dtr.dev.cray.com",
      "domain": "nmn mtl hmn",
      "first-master-hostname": "ncn-m002",
      "host_records": [
        {
          "aliases": [
            "ncn-w003.chn"
          ],
          "ip": "10.102.4.205"
        },
        {
          "aliases": [
            "ncn-w003.can"
          ],
          "ip": "10.102.4.141"
        },
        {
          "aliases": [
            "ncn-w003.cmn"
          ],
          "ip": "10.102.4.27"
        },
        {
          "aliases": [
            "ncn-w003.hmn"
          ],
          "ip": "10.254.1.20"
        },
        {
          "aliases": [
            "ncn-w003.mtl"
          ],
          "ip": "10.1.1.10"
        },
        {
          "aliases": [
            "ncn-w003.nmn",
            "ncn-w003"
          ],
          "ip": "10.252.1.12"
        },
        {
          "aliases": [
            "ncn-w003-mgmt"
          ],
          "ip": "10.254.1.19"
        },
        {
          "aliases": [
            "ncn-m001.can"
          ],
          "ip": "10.102.4.139"
        },
        {
          "aliases": [
            "ncn-m001.cmn"
          ],
          "ip": "10.102.4.25"
        },
        {
          "aliases": [
            "ncn-m001.hmn"
          ],
          "ip": "10.254.1.16"
        },
        {
          "aliases": [
            "ncn-m001.mtl"
          ],
          "ip": "10.1.1.8"
        },
        {
          "aliases": [
            "ncn-m001.nmn",
            "ncn-m001"
          ],
          "ip": "10.252.1.10"
        },
        {
          "aliases": [
            "ncn-m001.chn"
          ],
          "ip": "10.102.4.203"
        },
        {
          "aliases": [
            "ncn-m001-mgmt"
          ],
          "ip": "10.254.1.15"
        },
        {
          "aliases": [
            "ncn-w001.can"
          ],
          "ip": "10.102.4.143"
        },
        {
          "aliases": [
            "ncn-w001.cmn"
          ],
          "ip": "10.102.4.29"
        },
        {
          "aliases": [
            "ncn-w001.hmn"
          ],
          "ip": "10.254.1.24"
        },
        {
          "aliases": [
            "ncn-w001.mtl"
          ],
          "ip": "10.1.1.12"
        },
        {
          "aliases": [
            "ncn-w001.nmn",
            "ncn-w001"
          ],
          "ip": "10.252.1.14"
        },
        {
          "aliases": [
            "ncn-w001.chn"
          ],
          "ip": "10.102.4.207"
        },
        {
          "aliases": [
            "ncn-w001-mgmt"
          ],
          "ip": "10.254.1.23"
        },
        {
          "aliases": [
            "ncn-w004.can"
          ],
          "ip": "10.102.4.140"
        },
        {
          "aliases": [
            "ncn-w004.cmn"
          ],
          "ip": "10.102.4.26"
        },
        {
          "aliases": [
            "ncn-w004.hmn"
          ],
          "ip": "10.254.1.18"
        },
        {
          "aliases": [
            "ncn-w004.mtl"
          ],
          "ip": "10.1.1.9"
        },
        {
          "aliases": [
            "ncn-w004.nmn",
            "ncn-w004"
          ],
          "ip": "10.252.1.11"
        },
        {
          "aliases": [
            "ncn-w004.chn"
          ],
          "ip": "10.102.4.204"
        },
        {
          "aliases": [
            "ncn-w004-mgmt"
          ],
          "ip": "10.254.1.17"
        },
        {
          "aliases": [
            "ncn-s002.can"
          ],
          "ip": "10.102.4.135"
        },
        {
          "aliases": [
            "ncn-s002.cmn"
          ],
          "ip": "10.102.4.21"
        },
        {
          "aliases": [
            "ncn-s002.hmn"
          ],
          "ip": "10.254.1.8"
        },
        {
          "aliases": [
            "ncn-s002.mtl"
          ],
          "ip": "10.1.1.4"
        },
        {
          "aliases": [
            "ncn-s002.nmn",
            "ncn-s002"
          ],
          "ip": "10.252.1.6"
        },
        {
          "aliases": [
            "ncn-s002.chn"
          ],
          "ip": "10.102.4.199"
        },
        {
          "aliases": [
            "ncn-s002-mgmt"
          ],
          "ip": "10.254.1.7"
        },
        {
          "aliases": [
            "ncn-s004.can"
          ],
          "ip": "10.102.4.133"
        },
        {
          "aliases": [
            "ncn-s004.cmn"
          ],
          "ip": "10.102.4.19"
        },
        {
          "aliases": [
            "ncn-s004.hmn"
          ],
          "ip": "10.254.1.4"
        },
        {
          "aliases": [
            "ncn-s004.mtl"
          ],
          "ip": "10.1.1.2"
        },
        {
          "aliases": [
            "ncn-s004.nmn",
            "ncn-s004"
          ],
          "ip": "10.252.1.4"
        },
        {
          "aliases": [
            "ncn-s004.chn"
          ],
          "ip": "10.102.4.197"
        },
        {
          "aliases": [
            "ncn-s004-mgmt"
          ],
          "ip": "10.254.1.3"
        },
        {
          "aliases": [
            "ncn-s001.mtl"
          ],
          "ip": "10.1.1.5"
        },
        {
          "aliases": [
            "ncn-s001.nmn",
            "ncn-s001"
          ],
          "ip": "10.252.1.7"
        },
        {
          "aliases": [
            "ncn-s001.chn"
          ],
          "ip": "10.102.4.200"
        },
        {
          "aliases": [
            "ncn-s001.can"
          ],
          "ip": "10.102.4.136"
        },
        {
          "aliases": [
            "ncn-s001.cmn"
          ],
          "ip": "10.102.4.22"
        },
        {
          "aliases": [
            "ncn-s001.hmn"
          ],
          "ip": "10.254.1.10"
        },
        {
          "aliases": [
            "ncn-s001-mgmt"
          ],
          "ip": "10.254.1.9"
        },
        {
          "aliases": [
            "ncn-s003.can"
          ],
          "ip": "10.102.4.134"
        },
        {
          "aliases": [
            "ncn-s003.cmn"
          ],
          "ip": "10.102.4.20"
        },
        {
          "aliases": [
            "ncn-s003.hmn"
          ],
          "ip": "10.254.1.6"
        },
        {
          "aliases": [
            "ncn-s003.mtl"
          ],
          "ip": "10.1.1.3"
        },
        {
          "aliases": [
            "ncn-s003.nmn",
            "ncn-s003"
          ],
          "ip": "10.252.1.5"
        },
        {
          "aliases": [
            "ncn-s003.chn"
          ],
          "ip": "10.102.4.198"
        },
        {
          "aliases": [
            "ncn-s003-mgmt"
          ],
          "ip": "10.254.1.5"
        },
        {
          "aliases": [
            "ncn-m003.can"
          ],
          "ip": "10.102.4.137"
        },
        {
          "aliases": [
            "ncn-m003.cmn"
          ],
          "ip": "10.102.4.23"
        },
        {
          "aliases": [
            "ncn-m003.hmn"
          ],
          "ip": "10.254.1.12"
        },
        {
          "aliases": [
            "ncn-m003.mtl"
          ],
          "ip": "10.1.1.6"
        },
        {
          "aliases": [
            "ncn-m003.nmn",
            "ncn-m003"
          ],
          "ip": "10.252.1.8"
        },
        {
          "aliases": [
            "ncn-m003.chn"
          ],
          "ip": "10.102.4.201"
        },
        {
          "aliases": [
            "ncn-m003-mgmt"
          ],
          "ip": "10.254.1.11"
        },
        {
          "aliases": [
            "ncn-m002.chn"
          ],
          "ip": "10.102.4.202"
        },
        {
          "aliases": [
            "ncn-m002.can"
          ],
          "ip": "10.102.4.138"
        },
        {
          "aliases": [
            "ncn-m002.cmn"
          ],
          "ip": "10.102.4.24"
        },
        {
          "aliases": [
            "ncn-m002.hmn"
          ],
          "ip": "10.254.1.14"
        },
        {
          "aliases": [
            "ncn-m002.mtl"
          ],
          "ip": "10.1.1.7"
        },
        {
          "aliases": [
            "ncn-m002.nmn",
            "ncn-m002"
          ],
          "ip": "10.252.1.9"
        },
        {
          "aliases": [
            "ncn-m002-mgmt"
          ],
          "ip": "10.254.1.13"
        },
        {
          "aliases": [
            "ncn-w002.hmn"
          ],
          "ip": "10.254.1.22"
        },
        {
          "aliases": [
            "ncn-w002.mtl"
          ],
          "ip": "10.1.1.11"
        },
        {
          "aliases": [
            "ncn-w002.nmn",
            "ncn-w002"
          ],
          "ip": "10.252.1.13"
        },
        {
          "aliases": [
            "ncn-w002.chn"
          ],
          "ip": "10.102.4.206"
        },
        {
          "aliases": [
            "ncn-w002.can"
          ],
          "ip": "10.102.4.142"
        },
        {
          "aliases": [
            "ncn-w002.cmn"
          ],
          "ip": "10.102.4.28"
        },
        {
          "aliases": [
            "ncn-w002-mgmt"
          ],
          "ip": "10.254.1.21"
        },
        {
          "aliases": [
            "kubeapi-vip",
            "kubeapi-vip.nmn"
          ],
          "ip": "10.252.1.2"
        },
        {
          "aliases": [
            "rgw-vip",
            "rgw-vip.nmn"
          ],
          "ip": "10.252.1.3"
        },
        {
          "aliases": [
            "pit",
            "pit.nmn"
          ],
          "ip": "10.252.1.10"
        },
        {
          "aliases": [
            "packages.local",
            "registry.local"
          ],
          "ip": "10.92.100.71"
        },
        {
          "aliases": [
            "sw-spine-001"
          ],
          "ip": "10.252.0.2"
        },
        {
          "aliases": [
            "sw-spine-002"
          ],
          "ip": "10.252.0.3"
        },
        {
          "aliases": [
            "sw-leaf-bmc-001"
          ],
          "ip": "10.252.0.4"
        }
      ],
      "internal-domain": "~FIXME~",
      "k8s-api-auditing-enabled": false,
      "k8s-virtual-ip": "10.252.1.2",
      "kubernetes-max-pods-per-node": "200",
      "kubernetes-pods-cidr": "10.32.0.0/12",
      "kubernetes-services-cidr": "10.16.0.0/12",
      "kubernetes-weave-mtu": "1376",
      "ncn-mgmt-node-auditing-enabled": false,
      "num_storage_nodes": 4,
      "rgw-virtual-ip": "10.252.1.3",
      "site-domain": "dev.cray.com",
      "spire": {
        "fqdn": "10.92.100.82",
        "trustdomain": "shasta"
      },
      "system-name": "fanta",
      "wipe-ceph-osds": "yes"
    },
    "user-data": null,
    "phone-home": {
      "pub_key_dsa": "",
      "pub_key_rsa": "",
      "pub_key_ecdsa": "",
      "instance_id": "",
      "hostname": "",
      "fqdn": ""
    }
  }
}
```

**Comment by rsjostrand-hpe** [comment]
Here is what the updated boot parameters for x3000c0s17b0n0/ncn-w003 look like out of the unit test:
```
{
  "hosts": [
    "x3000c0s17b0n0"
  ],
  "params": "ifname=mgmt1:b8:59:9f:34:88:7b ifname=mgmt0:b8:59:9f:34:88:7a biosdevname=1 pcie_ports=native transparent_hugepage=never console=tty0 console=ttyS0,115200 iommu=pt metal.server=http://rgw-vip.nmn/ncn-images/ceph/0.1.113 metal.no-wipe=1 ds=nocloud-net;s=http://10.92.100.81:8888/ rootfallback=LABEL=BOOTRAID initrd=initrd.img.xz root=live:LABEL=SQFSRAID rd.live.ram=0 rd.writable.fsimg=0 rd.skipfsck rd.live.squashimg=filesystem.squashfs rd.live.overlay=LABEL=ROOTRAID rd.live.overlay.thin=1 rd.live.overlay.overlayfs=1 rd.luks rd.luks.crypttab=0 rd.lvm.conf=0 rd.lvm=1 rd.auto=1 rd.md=1 rd.dm=0 rd.neednet=0 rd.peerdns=1 rd.md.waitclean=1 rd.multipath=0 rd.md.conf=1 rd.bootif=0 hostname=ncn-s003 rd.net.timeout.carrier=120 rd.net.timeout.ifup=120 rd.net.timeout.iflink=120 rd.net.dhcp.retry=3 rd.net.timeout.ipv6auto=0 rd.net.timeout.ipv6dad=0 append nosplash quiet crashkernel=360M log_buf_len=1 rd.retry=10 rd.shell ip=mgmt0:dhcp",
  "kernel": "s3://ncn-images/ceph/0.1.113/kernel",
  "initrd": "s3://ncn-images/ceph/0.1.113/initrd",
  "cloud-init": {
    "meta-data": {
      "availability-zone": "x3000",
      "instance-id": "i-B91E06E1",
      "ipam": {
        "can": {
          "gateway": "10.102.4.129",
          "ip": "10.102.4.134/26",
          "parent_device": "bond0",
          "vlanid": 6
        },
        "cmn": {
          "gateway": "10.102.4.1",
          "ip": "10.102.4.20/25",
          "parent_device": "bond0",
          "vlanid": 7
        },
        "hmn": {
          "gateway": "10.254.0.1",
          "ip": "10.254.1.6/17",
          "parent_device": "bond0",
          "vlanid": 4
        },
        "mtl": {
          "gateway": "10.1.0.1",
          "ip": "10.1.1.3/16",
          "parent_device": "bond0",
          "vlanid": 0
        },
        "nmn": {
          "gateway": "10.252.0.1",
          "ip": "10.252.1.5/17",
          "parent_device": "bond0",
          "vlanid": 2
        }
      },
      "local-hostname": "ncn-s003",
      "region": "fanta",
      "shasta-role": "ncn-storage",
      "xname": "x3000c0s17b0n0"
    },
    "user-data": {
      "hostname": "ncn-s003",
      "local_hostname": "ncn-s003",
      "mac0": {
        "gateway": "10.252.0.1",
        "ip": "",
        "mask": "10.252.2.0/23"
      },
      "ntp": {
        "allow": [
          "10.92.100.0/24",
          "10.102.4.0/25",
          "10.252.0.0/17",
          "10.253.0.0/16",
          "10.1.1.0/16",
          "10.254.0.0/17",
          "10.107.0.0/17",
          "10.106.0.0/17",
          "10.94.100.0/24"
        ],
        "config": {
          "confpath": "/etc/chrony.d/cray.conf",
          "template": "## template: jinja\n# csm-generated config.  Do not modify--changes can be overwritten\n{% for pool in pools | sort -%}\n{% if local_hostname == 'ncn-m001' and pool == 'ncn-m001' %}\n{% endif %}\n{% if local_hostname != 'ncn-m001' and pool != 'ncn-m001' %}\n{% else %}\npool {{ pool }} iburst\n{% endif %}\n{% endfor %}\n{% for server in servers | sort -%}\n{% if local_hostname == 'ncn-m001' and server == 'ncn-m001' %}\n{% endif %}\n{% if local_hostname != 'ncn-m001' and server != 'ncn-m001' %}\n{% else %}\nserver {{ server }} iburst trust\ninitstepslew 1 {{ server }}\n{% endif %}\n{% endfor %}\n{% for peer in peers | sort -%}\n{% if local_hostname == peer %}\n{% else %}\n{% if loop.index \u003c= 9 %}\n{# Only add 9 peers to prevent too much NTP traffic #}\npeer {{ peer }} minpoll -2 maxpoll 9 iburst\n{% endif %}\n{% endif %}\n{% endfor %}\n{% for net in allow | sort -%}\nallow {{ net }}\n{% endfor %}\n{% if local_hostname == 'ncn-m001' %}\nlocal stratum 8 orphan\n{% else %}\nlocal stratum 10 orphan\n{% endif %}\nlog measurements statistics tracking\nlogchange 1.0\nmakestep 0.1 3\n"
        },
        "enabled": true,
        "ntp_client": "chrony",
        "peers": [
          "ncn-m001",
          "ncn-m002",
          "ncn-m003",
          "ncn-w001",
          "ncn-w002",
          "ncn-w003",
          "ncn-s001",
          "ncn-s002",
          "ncn-s003"
        ],
        "pools": [
          "time.nist.gov"
        ],
        "servers": [
          "ncn-m001"
        ]
      },
      "runcmd": [
        "/srv/cray/scripts/metal/net-init.sh",
        "/srv/cray/scripts/common/update_ca_certs.py",
        "/srv/cray/scripts/metal/install.sh",
        "/srv/cray/scripts/common/pre-load-images.sh",
        "touch /etc/cloud/cloud-init.disabled",
        "/srv/cray/scripts/common/ceph-enable-services.sh"
      ],
      "timezone": "UTC",
      "write_files": [
        {
          "content": "10.94.100.0/24 10.254.0.1 - bond0.hmn0\n10.107.0.0/22 10.254.0.1 - bond0.hmn0",
          "owner": "root:root",
          "path": "/etc/sysconfig/network/ifroute-bond0.hmn0",
          "permissions": "0644"
        },
        {
          "content": "10.252.2.0/23 10.252.0.1 - bond0.nmn0\n10.92.100.0/24 10.252.0.1 - bond0.nmn0\n10.106.0.0/22 10.252.0.1 - bond0.nmn0",
          "owner": "root:root",
          "path": "/etc/sysconfig/network/ifroute-bond0.nmn0",
          "permissions": "0644"
        },
        {
          "content": "10.102.4.60/30 10.102.4.1 - bond0.cmn0\n10.102.4.64/26 10.102.4.1 - bond0.cmn0",
          "owner": "root:root",
          "path": "/etc/sysconfig/network/ifroute-bond0.cmn0",
          "permissions": "0644"
        }
      ]
    },
    "phone-home": {
      "pub_key_dsa": "",
      "pub_key_rsa": "",
      "pub_key_ecdsa": "",
      "instance_id": "",
      "hostname": "",
      "fqdn": ""
    }
  }
}
```

File: cmd/upgrade-metadata.go [hunk 1]
```diff
@@ -186,6 +187,121 @@ func getWriteFiles(networks sls_common.NetworkArray, ipamNetworks bss.CloudInitI
...
+		buildBSSHostRecords(networkEPs, "NMNLB", "nmn_metallb_address_pool", "istio-ingressgateway", []string{"packages.local", "registry.local"}),
+	)
+
+	// Add entries for switches
+	nmnNetSubnet, err := networkEPs["NMN"].LookupSubnet("network_hardware")
+	if err != nil {
+		log.Fatal("Unable to find network_hardware in the NMN network")
+	}
+
+	for _, ipResveration := range nmnNetSubnet.IPReservations {
```
Comment (johren-hpe) [inline cmd/upgrade-metadata.go:293]: Just a typo but doesn't affect the functionality since it carries through:
   ipResveration -> ipReservation

**Review by johren-hpe — APPROVED** [review]

**Comment by johren-hpe** [comment]
Looks like three of the files need licenses added:
```
Warning:  License is not detected: cmd/update-metadata_test.go
[INFO] Scanning file cmd/upgrade-metadata.go
Warning:  License is not detected: cmd/upgrade-metadata.go
[INFO] Scanning file pkg/bss/types.go
Warning:  License is not detected: pkg/bss/types.go
```

File: cmd/upgrade-metadata.go (hunk 1 above)
Comment (rsjostrand-hpe) [reply cmd/upgrade-metadata.go]: Fixed

**Pull request merged**